app/src/main/kotlin/de/digitalService/useID/ui/coordinators/AppCoordinator.kt

```diff
@@ -54,26 +64,34 @@ class AppCoordinator @Inject constructor(
         coldLaunch = false
     }
 
-    override fun setNfcAvailability(availability: NfcAvailability) {
-        nfcAvailability.value = availability
-    }
-
     override fun handleDeepLink(uri: Uri) {
         logger.debug("Handling deep link.")
 
         Uri.parse(uri.toString()).getQueryParameter("tcTokenURL")?.let { url ->
-            tcTokenUrl = url
-
-            navigator.popToRoot()
-            if (storageManager.firstTimeUser) {
-                offerIdSetup(url)
-            } else {
-                identificationCoordinator.startIdentificationProcess(url, false)
-            }
-
-            coldLaunch = false
+            handleTcTokenUrl(url)
         } ?: run {
             logger.info("URL does not contain tcTokenUrl parameter.")
         }
     }
+
+    private fun handleTcTokenUrl(url: String) {
+        navigator.popToRoot()
+
+        if (nfcInterfaceManager.nfcAvailability.value != NfcAvailability.Available) {
+            logger.debug("Blocking identification due to unavailable NFC.")
+            cachedTcTokenUrl = url
+            return
+        } else {
+            cachedTcTokenUrl = null
```

Comment:
I am not sure if I misunderstand this, but shouldn't the two lines be switched? If Nfc is not available -> set the cachedTcTokenUrl to null?

Reply:
If NFC is not available we want to cache the URL for later (after it has been enabled again eventually). If it is then enabled later the cached URL should be reset to null.